### ISSUE TITLE
Php4ConstructorFixer causes internal error

### ISSUE THREAD

**Issue by stefanotorresi** [FriendsOfPHP/PHP-CS-Fixer#1197]
this filter, with some files only, causes `Symfony\CS\Tokenizer\Tokens` to throw at [this line](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/108a513466b784253bc6a1fe9bbdb1851ea9503b/Symfony/CS/Tokenizer/Tokens.php#L348)

reproduced in both 1.8 and master.

**Comment by keradus** [comment]
Could you provide example on what code fixers crashes ?

**Comment by stefanotorresi** [comment]
these files

```
Files that were not fixed due to internal error:
   1) Storage/DataMapperAdapter.php
   2) Options/ModuleOptions.php
   3) Entity/Client.php
   4) Entity/Scope.php
   5) Entity/AuthorizationCode.php
   6) Entity/AccessToken.php
   7) Entity/User.php
   8) Entity/RefreshToken.php
   9) Entity/AbstractToken.php
  10) Entity/ThirdParty.php
  11) GrantType/UserCredentials/PasswordStrategy.php
  12) GrantType/ThirdPartyGrantType.php
  13) GrantType/ThirdParty/Provider/FacebookProvider.php
  14) GrantType/ThirdParty/Provider/ProviderFactory.php
  15) GrantType/ThirdParty/Provider/InstagramProvider.php
  16) GrantType/ThirdParty/Provider/Exception/ClientException.php
  17) GrantType/ThirdParty/Provider/Exception/DomainException.php
  18) GrantType/ThirdParty/Provider/Exception/InvalidArgumentException.php
  19) Server/ServerInitializer.php
  20) Factory/ModuleOptionsFactory.php
  21) Factory/ThirdPartyGrantTypeFactory.php
  22) Factory/DataMapperAdapterFactory.php
  23) Factory/BcryptFactory.php
  24) tests/Storage/DataMapperAdapterTest.php
  25) tests/Server/ServerInitializerTest.php
  26) tests/Asset/ScopeAwareUser.php
  27) Module.php
```

at [stefanotorresi/thorr-oauth2](https://github.com/stefanotorresi/thorr-oauth2/tree/3bce8e5e675c2a656a3b69486fa448c835d97f43)

**Comment by gharlan** [comment]
I can not reproduce it (with your code). 
Which PHP version?

**Comment by stefanotorresi** [comment]
> I can not reproduce it (with your code). 
> Which PHP version?

i reproduced this with 5.4.37, 5.5.24 and 5.6.5.
note that the fixer is not enabled by default in the project i linked.

**Comment by gharlan** [comment]
hmm, so how can we reproduce it?
What I did:

```
$ git clone https://github.com/stefanotorresi/thorr-oauth2/tree/3bce8e5e675c2a656a3b69486fa448c835d97f43 thorr-oauth2
$ cd thorr-oauth2
$ php-cs-fixer fix
```

Result is:

```
Loaded config from "[...]/thorr-oauth2/.php_cs"
Fixed all files in 7.407 seconds, 13.000 MB memory used
```

**Comment by stefanotorresi** [comment]
as i said, the fixer is not enabled by default, so you have to enable it explicitly:

`php-cs-fixer fix -v --dry-run --fixers php4_constructor`

**Comment by gharlan** [comment]
Ah of course, sorry! ;)

**Comment by keradus** [comment]
Closing due to merge of #1201
Fix can be found at v1.8.1@dev
// cc @stefanotorresi
